File: sdk/ios/KnoemaMobile/Sources/KnoemaMobile/NPCAgent.swift
import Foundation
#if canImport(FoundationNetworking)
import FoundationNetworking
#endif

public struct NPCAgent: Sendable {
    public let agentId: String
    public let displayName: String
    public let client: KnoemaClient

    public init(agentId: String, displayName: String, client: KnoemaClient) {
        self.agentId = agentId
        self.displayName = displayName
        self.client = client
    }

    public func cachedOrOfflineResponse(input: String) -> NPCActionResponse {
        let key = "\(agentId):\(input)"
        if let cached = client.cachedResponse(for: key) {
            return cached
        }
        let response = client.offlineResponse(agentId: agentId, input: input)
        client.cache(response: response, for: key)
        return response
    }

    public func agentEndpointRequest() -> URLRequest {
        client.buildRESTRequest(path: "/agents/\(agentId)", method: "GET")
    }
}

File: sdk/android/luvoire-mobile/src/main/kotlin/com/celovin/luvoire/NPCAgent.kt
package com.celovin.luvoire

class NPCAgent(
    private val agentId: String,
    private val displayName: String,
    private val client: LuvoireClient,
) {
    fun endpoint(): String {
        return client.restUrl("/agents/$agentId")
    }

    fun name(): String {
        return displayName
    }

    fun cachedOrOfflineResponse(input: String): NPCActionResponse {
        val key = "$agentId:$input"
        val cached = client.cachedResponse(key)
        if (cached != null) {
            return cached
        }
        val response = client.offlineResponse(agentId, input)
        client.cacheResponse(key, response)
        return response
    }
}
